packages/create-ponder/src/helpers/graphqlTranslator.ts

```diff
@@ -0,0 +1,265 @@
+import { gql } from "graphql-tag";
+
+import type {
+  DefinitionNode,
+  NamedTypeNode,
+  NonNullTypeNode,
+} from "graphql/language";
+
+const GraphqlTypesToPonderTypes = {
+  ID: "string",
+  String: "string",
+  Int: "int",
+  BigInt: "bigint",
+  BigDecimal: "float",
+  Float: "float",
+  Boolean: "boolean",
+  Bytes: "hex",
+};
+
+export function translateSchema(schema: string): string {
+  const schemaAST = gql(schema);
+  const result: string[] = [];
+  const enums: string[] = [];
+  const enumsResult: string[] = [];
+  const idTypes: { [table: string]: string } = {};
+  const referenceFields: { [table: string]: { [key: string]: string } } = {};
+  const tables: {
+    [table: string]: {
+      columns: { [key: string]: string };
+      descriptions: { [key: string]: string };
+    };
+  } = {};
+  const joinedTables: { [name: string]: string } = {};
+
+  // Parse enums
+  // Need to do all the enums first before processing other types
+  // because there is no other way to determine if a graphql definition is referring to
+  // an enum or another table
+  Object.keys(schemaAST.definitions).forEach((definitionName) => {
+    const definition = schemaAST.definitions[
+      definitionName as any as number
+    ] as DefinitionNode;
+    if (definition.kind === "EnumTypeDefinition") {
+      const enumName = definition.name.value;
+      enums.push(enumName);
+      const values: string[] = [];
+      definition.values?.forEach((value) => {
+        values.push(value.name.value);
+      });
+
+      enumsResult.push(
+        `	${enumName}: p.createEnum(${JSON.stringify(values)}),\n\n`,
+      );
+    }
+  });
```

Comment:
Just a style nit - I'd write this as
```ts
// Build enums first because they are required to determine if a
// column type references an enum or another table.
for (const definition of schemaAST.definitions) {
  if (definition.kind === "EnumTypeDefinition") {
    const name = definition.name.value;
    const values = definition.values?.map((value) => value.name.value) ?? [];
    enums.push(name);
    enumsResult.push(`${name}: p.createEnum(${JSON.stringify(values)})`);
  }
}
```